talk.tex

```diff
@@ -52,20 +52,77 @@
 \frametitle{Introduction}
 \framesubtitle{Goals}
 \begin{itemize}
-\item Be the fastest in symbolic manipulation (opensource or commercial)
-\item Serve as the core for SymPy
+\item Be the fastest symbolic manipulation library (opensource or commercial)
+\item Serve as the core for SymPy and Sage
+\item Serve as the default symbolic manipulation library in other languages
+    thanks to thin wrappers (Python, Ruby, Julia, ...)
 \end{itemize}
 \end{frame}
 
 
 \begin{frame}
-\frametitle{Why C++}
+\frametitle{What language to use}
+\begin{block}{Problem}
+SymPy speed is sometimes insufficient
 \begin{itemize}
-\item SymPy speed is insufficient to handle very large expressions.
+\item Handling of very large expressions
+\item Large calculations using small/medium size expressions
+\end{itemize}
+\end{block}
+
+\begin{block}{Let's fix that}
+\begin{itemize}
+\item We tried: pure Python/PyPy, Cython, C
+\item Investigated Julia, Rust, Scala, Javascript, ...
+\item Chose C++
+\end{itemize}
+\end{block}
+\end{frame}
+
+
+\begin{frame}
+\frametitle{Why Pure C++}
+\begin{itemize}
+\item \textbf{Fast} in Release mode, but safe in Debug mode
+\item Compiler helps (not as good as Scala or Haskell, but much better than
+    Python)
+\item Just one language to learn, thus easy to maintain (as opposed to several
+    intertwined layers such as C + Cython + Python)
+\item Thin wrappers (that core developers do not need to maintain), all functionality in C++
 \item Easier to create bindings to other languages like Python, Julia, Ruby and Haskell.
 \end{itemize}
 \end{frame}
 
+\begin{frame}
+\frametitle{Why Pure C++: Fast in Release Mode}
+\begin{itemize}
+    \item Allows direct memory handling (allocation, deallocation, access)
+    \item Allows to tweak how and when things are done
+    \item It is possible to go to bare metal
```

Comment:
I think so. It's not easily possible to go to bare metal with Cython for
example, because if you use dicts, it uses Python dicts.

Sent from my mobile phone.
On Jul 4, 2016 10:42 PM, "Isuru Fernando" <notifications@github.com> wrote:

> In talk.tex
> <https://github.com/certik/scipy-2016-symengine-talk/pull/16#discussion_r69507186>
> :
>
> >  \item Easier to create bindings to other languages like Python, Julia, Ruby and Haskell.
> >  \end{itemize}
> >  \end{frame}
> >
> > +\begin{frame}
> > +\frametitle{Why Pure C++: Fast in Release Mode}
> > +\begin{itemize}
> > +    \item Allows direct memory handling (allocation, deallocation, access)
> > +    \item Allows to tweak how and when things are done
> > +    \item It is possible to go to bare metal
>
> Is this item necessary?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/certik/scipy-2016-symengine-talk/pull/16/files/a532b8949a8dbd75584821c6e1fb82099e1dddff#r69507186>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABQWEwKmKe1tIExdzvYHge1y7jytJHzks5qSeCwgaJpZM4JD_2i>
> .
>